Add CLI interface - Superseded

benny779 opened the pull request:
closes #19

benny779 committed:
feat: add CLI interface

benny779 committed:
refactor: move utility functions to utils folder

benny779 committed:
refactor: extract error and FPI utility functions

benny779 committed:
fix: add descriptions to registry-url and cache-path options

mechanik-daniel commented:
* The bin path is incorrect, according to the ncc build script the output will be dist/cli/index.js
* You should add the --external flag, with the same external modules as the main build script
* Need to add the "shebang" line at the top of the cli script, and make sure ncc respects it and adds it to the top of the output index.js file
* Please update the readme and document the CLI arguments

benny779 committed:
build: change CLI build location

benny779 committed:
Merge branch 'Outburn-IL:main' into cli

benny779 committed:
build: add shebang to cli

benny779 committed:
docs: add CLI documentation

benny779 committed:
test: add ability to skip heavy tests via environment variable (fix #25)

mechanik-daniel commented on src/commander.d.ts:
Why is this file necessary? The new dependency @commander-js/extra-typings should be imported directly where it's used, no need for this wrapper file

mechanik-daniel commented on src/index.ts:
```diff
@@ -531,7 +445,7 @@ export class FhirPackageInstaller {
         } catch (e) {
           this.logger.error(`Failed to install package ${packageObject.id}@${packageObject.version}`);
-          throw this.prethrow(e);
+          throw e;
```
Why `throw this.prethrow(e)` -> `throw e`?

mechanik-daniel commented on src/index.ts:
```diff
@@ -661,7 +563,7 @@ export class FhirPackageInstaller {
       } catch (e) {
         this.logger.error(`Failed to download package ${packageName}`);
-        throw this.prethrow(e);
+        throw e;
```
same

benny779 committed:
Merge branch 'main' of https://github.com/Outburn-IL/fhir-package-installer into cli

benny779 committed:
fix: use cross-env to set environment variable for skipping heavy tests

benny779 commented on src/index.ts:
```diff
@@ -454,7 +374,7 @@ export class FhirPackageInstaller {
         } catch (e) {
           this.logger.error(`Failed to fetch latest version for package ${packageName}`);
-          throw this.prethrow(e);
+          throw e;
```
same

benny779 committed:
fix: update commander import to use @commander-js/extra-typings

benny779 committed:
fix: correct prethrow assignment order in FhirPackageInstaller constructor

benny779 committed:
refactor: simplify finalPath assignment logic in FhirPackageInstaller

benny779 committed:
fix: enhance descriptions for download and install-local commands in CLI

benny779 committed:
chore: add package-lock.json to .gitignore

benny779 committed:
Merge branch 'main' of https://github.com/Outburn-IL/fhir-package-installer into cli

benny779 committed:
fix: move again http utils to a dedicated file after merge

benny779 committed:
fix: return back package-lock.json to support 'npm ci' (for github actions)

benny779 committed:
refactor: consolidate HTTP utilities and centralize auth handling

- Create shared makeHttpRequest function to eliminate duplication
- Move 401/403 error handling to shared layer for consistency
- Reduce ~75 lines of duplicate code between fetchJson/fetchStream

benny779 committed:
fix: improve race condition handling during package installation

benny779 committed:
fix: update log message to reflect package index generation in cache

benny779 committed:
fix: add race-test-cache to .gitignore

benny779 committed:
fix: make fpi cli working

benny779 committed:
chore: update CLI version to 1.0.0

benny779 committed:
fix: add error handling to cli commands using Proxy

benny779 committed:
fix: add success message after package installation

benny779 committed:
Merge branch 'main' of https://github.com/Outburn-IL/fhir-package-installer into cli
